authentication-typescript/template/App.tsx

```diff
@@ -44,7 +44,7 @@ export const App = (): JSX.Element => {
     const { getInitialState } = useLinking(ref, authLinkMapping);
     const [initialState, setInitialState] = React.useState();
     React.useEffect(() => {
-        resolveInitialState(getInitialState, setInitialState);
+        resolveInitialState(getInitialState as () => any, setInitialState as (state: any) => void);
```

Comment:
We shouldn't be using the `any` type

Reply:
This is a little tricky; `getInitialState` is the correct type already (even without the `any` cast) but the auth-ts template fails linting because it doesn't have access to the type definitions found within `@react-navigation/native`. 

I'm confused as to why we're creating templates without having the required deps installed at the root level.

Reply:
I think this story is going to evolve into something larger than expected; how is linting working without the type dependencies installed?

Reply:
@daileytj do you have thoughts on this one?  We might need to update the root package.json so it contains all the dependencies that each template requires; this is how the angular one does it currently.

Reply:
The root package.json absolutely should contain every dependency that is in any template

Reply:
This is how it was initially written so that the demo project could run every template locally for testing purposes